src/Hcal/HcalTriggerGeometry.cxx

```diff
@@ -0,0 +1,228 @@
+    //       ldmx::HcalTriggerID tid(0, 0, tcell);
+    //       std::vector<ldmx::HcalID> pids;
+    //       for (int dv = -1; dv <= 1; dv++) {
+    //         for (int du = -1; du <= 1; du++) {
+    //           ldmx::HcalID pid(0, 0, u + du + dv,
+    //                            v + dv);  // changes directions here
+    //           precision2trigger_[pid] = tid;
+    //           pids.push_back(pid);
+    //         }
+    //       }
+    //       trigger2precision_[tid] = pids;
+    //       tcell++;
+    //     }
+    //   }
+    //   // right side
+    //   for (int v = 2; v <= 22; v += 3) {
+    //     int irow = (v - 2) / 3;
+    //     for (int icol = 0; icol <= std::min(irow, 3); icol++) {
+    //       if (irow - icol >= 4) continue;
+    //       ldmx::HcalTriggerID tid(0, 0, tcell);
+    //       std::vector<ldmx::HcalID> pids;
+    //       int u = 13 + 3 * icol;
+    //       for (int dv = -1; dv <= 1; dv++) {
+    //         for (int du = -1; du <= 1; du++) {
+    //           ldmx::HcalID pid(0, 0, u + du, v + du + dv);
+    //           precision2trigger_[pid] = tid;
+    //           pids.push_back(pid);
+    //         }
+    //       }
+    //       trigger2precision_[tid] = pids;
+    //       tcell++;
+    //     }
+    //   }
+
+    // } else {
+    //   // raise an exception...
+    // }
+}
+
+// std::vector<ldmx::HcalID> HcalTriggerGeometry::contentsOfTriggerCell(
+//     ldmx::HcalTriggerID triggerCell) const {
+//   ldmx::HcalTriggerID effId;
+//   std::vector<ldmx::HcalID> retval;
+//   // if ((symmetry_ & MODULES_MASK) == INPLANE_IDENTICAL) {
+//   //   effId = ldmx::HcalTriggerID(0, 0, triggerCell.triggercell());
+//   // }
+//   // auto ptr = trigger2precision_.find(effId);
+//   // if (ptr != trigger2precision_.end()) {
+//   //   for (auto idz : ptr->second) {
+//   //     retval.push_back(
+//   //         ldmx::HcalID(triggerCell.layer(), triggerCell.module(), idz.cell()));
+//   //   }
+//   // }
+//   return retval;
+// }
+
+// ldmx::HcalID HcalTriggerGeometry::centerInTriggerCell(
+//     ldmx::HcalTriggerID triggerCell) const {
+//   // ldmx::HcalTriggerID effId;
+//   // if ((symmetry_ & MODULES_MASK) == INPLANE_IDENTICAL) {
+//   //   effId = ldmx::HcalTriggerID(0, 0, triggerCell.triggercell());
+//   // }
+//   // auto ptr = trigger2precision_.find(effId);
+//   // if (ptr == trigger2precision_.end()) {
+//   //   std::stringstream ss;
+//   //   ss << "Unable to find trigger cell " << triggerCell;
+//   //   EXCEPTION_RAISE("HcalGeometryException", ss.str());
+//   // }
+
+//   // return ldmx::HcalID(triggerCell.layer(), triggerCell.module(),
+//   //                     ptr->second[4].cell());
+//     return ldmx::HcalID();
+// }
+
+ldmx::HcalTriggerID HcalTriggerGeometry::belongsTo(
+    ldmx::HcalID precisionCell) const {
+  unsigned int sstrip=0;
+  unsigned int strip=precisionCell.strip();
+  if (precisionCell.section() == ldmx::HcalID::HcalSection::BACK){
```

Comment:
This adapts the grouping presented by Martin Meier at the SWAN meeting on 19 July 2021.